source/lib/JMSLab/builders/build_lyx.py

```diff
@@ -26,19 +29,142 @@ def build_lyx(target, source, env):
+        Generate handout pdf if handout path exists in target and handout path
+        has the correct suffix and/or extension. Raise value error if intended 
+        behavior implied by target list contradicts behavior implied by 
+        HANDOUT_SFIX.
+        
+        Always generate main pdf.        
+        '''
+        
+        target_list = misc.make_list_if_string(target)      
+        
+        if target_list[0] in target_list[1:]:
+            raise ValueError(
+                'Error: main pdf path cannot be the same as handout pdf path')
+            
+            
+        if len(target_list) == 1:
+             if bool(env['HANDOUT_SFIX']):
+                 raise ValueError(
+                     'Error: HANDOUT_SFIX non-empty but no handout pdf path')
+        
+        elif len(target_list) > 1:
+
+            handout_flag = env['HANDOUT_SFIX'] + '.pdf'
+            handout_target_list = [x for x in target_list[1:]
+                                   if str(x).lower().endswith(handout_flag.lower())]
+                
+            if not bool(handout_target_list):
+                raise ValueError('Error: no valid targets for handout.')
+            
+            self.main_target = target_list[0]
+            self.handout_target_list = handout_target_list
+            self.create_handout()
+
+            traceback = ''
+            raise_system_call_exception = False
+            try:
+                subprocess.check_output(self.handout_call, 
+                                        shell = True, 
+                                        stderr = subprocess.STDOUT)
+            except subprocess.CalledProcessError as ex:
+                traceback = ex.output
+                raise_system_call_exception = True
+                
+            self.cleanup_handout()
+            if raise_system_call_exception:
+                self.raise_system_call_exception(traceback = traceback)
+        else:
+            pass
+        
+
+        traceback = ''
+        raise_system_call_exception = False
+        try:
+            subprocess.check_output(self.system_call, 
+                                    shell = True, 
+                                    stderr = subprocess.STDOUT)
+        except subprocess.CalledProcessError as ex:
+            traceback = ex.output
+            raise_system_call_exception = True
+
+        self.cleanup()
+        if raise_system_call_exception:
+            self.raise_system_call_exception(traceback = traceback)
+            
+        return None
+    
+    
+    def execute_system_call(self, target, env):
+        '''
+        Execute the system call attribute.
+        Log the execution.
+        Check that expected targets exist after execution.
+        '''
+        self.check_code_extension()
+        self.start_time = misc.current_time()
+        self.do_call(target, env)
+        self.check_targets()
+        self.timestamp_log(misc.current_time())
+        return None
+    
```

Comment:
Github is still not recognizing the extra line here. :-)

![image](https://user-images.githubusercontent.com/7840353/180876689-c9847eaa-eca8-4011-8932-ee99adfc6391.png)

Reply:
Thanks for catching that! Fixed now.